### PR TITLE
Initial Version of Card Search filter

### PR DESCRIPTION
**Pull request by plural**

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Holy shit, this is very cool. This parsing library is way better than my handcrafted parser I built with StringScanner way back in 2019.

I'm unbelievably happy to see y'all developing this, so glad using Ruby on Rails is proving to be a good choice. ❤️❤️❤️

**Comment by plural** [comment]
Thanks, @NoahTheDuke !  I'm quite happy with parslet so far.  Plenty left to do to make the search API we actually wanna see, but this is a really encouraging start.

File: lib/search_query_builder.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                    where << '%%%s%%' % value
+                end
+            end
+
+            # bare/quoted words in the query are automatically mapped to stripped_title
+            if f.include?(:string)
+                    value = f[:string].to_s
+                    operator = value.start_with?('!') ? 'NOT LIKE' : 'LIKE'
+                    value    = value.start_with?('!') ? value[1..] : value
+                    constraints << 'lower(stripped_title) %s ?' % operator
```
Comment (NoahTheDuke) [inline lib/search_query_builder.rb:161]: Is `lower` correct here? Will it match if given a capitalized term?

Comment (plural) [reply lib/search_query_builder.rb]: not without altering the input values, which i've done now.  Thanks for the nudge.

**Review by NoahTheDuke — APPROVED** [review]
This looks great. A solid base for iteration.

**Pull request merged**